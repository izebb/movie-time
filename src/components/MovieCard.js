import React from 'react';
import styled from 'styled-components';
import { imageSizes } from '../config';

import { MovieCardSummary } from './MovieCardSummary';
import { MovieImage } from './MovieImage';

const { card } = imageSizes;

const Container = styled.div`
  position: relative;
  width: ${card.width}px;
  height: ${card.height}px;
  cursor: pointer;
  overflow: hidden;
  border-radius: 5px;
`;

export const MovieCard = ({ id, title, seasons, coverTemplate, onClick }) => {
  const handleClick = () => {
    onClick(id);
  };

  return (
    <Container onClick={handleClick} data-testid="MovieCard">
      <MovieImage
        width={card.width}
        height={card.height}
        template={coverTemplate}
      />
      <MovieCardSummary title={title} seasons={seasons} />
    </Container>
  );
};
