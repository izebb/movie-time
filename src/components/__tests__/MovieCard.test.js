import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MovieCard } from '../MovieCard';

describe('MovieCard', () => {
  it('should render without crashing', () => {
    const props = {
      id: 'movie',
      title: 'a beautiful mind',
      seasons: 5,
      coverTemplate:
        'https://i-viaplay-com.akamaized.net/viaplay-prod/286/700/LethalWeapon3Exc_Packshot.jpg{?width,height}',
      onClick: jest.fn()
    };
    const wrapper = shallow(<MovieCard {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
