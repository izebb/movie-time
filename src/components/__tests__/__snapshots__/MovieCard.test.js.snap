// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render without crashing 1`] = `
<MovieCard__Container
  data-testid="MovieCard"
  onClick={[Function]}
>
  <MovieImage
    height={170}
    template="https://i-viaplay-com.akamaized.net/viaplay-prod/286/700/LethalWeapon3Exc_Packshot.jpg{?width,height}"
    width={270}
  />
  <MovieCardSummary
    seasons={5}
    title="a beautiful mind"
  />
</MovieCard__Container>
`;
